chain/client/src/client.rs

```diff
@@ -746,6 +747,19 @@ impl Client {
             )
         };
 
+        // Record a timestamp of processing a block, but only do it if the block isn't known yet.
+        let block_known_error = if let Err(ref e) = result {
+            match e.kind() {
+                ErrorKind::BlockKnown(_) => true,
+                _ => false,
+            }
+        } else {
+            false
+        };
+        if !block_known_error {
+            self.chunks_delay_tracker.received_block(&block_hash, processing_started);
```

Comment:
Good catch! Although I think this fix doesn't fix the issue completely, there could be other errors such as `EpochOutOfBounds` or if the block is invalid that will cause the same issue. And this change changes the timestamp for receiving block to after the block is processed. I think a better way to do this is to still call `received_block` at the beginning of this function, but remove the block hash if result is not Ok, or Error::Orphan or Error::ChunksMissing.

Reply:
You're right, thanks.